crowbar_framework/app/models/proposal.rb

```diff
@@ -87,6 +92,27 @@ def raw_data=(value)
 
   private
 
+  # XXX: we need to be careful to not create an endless loop
+  def update_corresponding_proposal_object
+    proposal_object = ProposalObject.find_proposal_by_id(self.key)
+    if proposal_object
+      if proposal_object.raw_data != self.raw_data
+        increment_crowbar_revision! if proposal_object.crowbar_revision >= self.crowbar_revision
```

Comment:
Ok, you were right. The `if` for revision compare is not needed. I dropped it. Still not sure about the 'always increment', though. Where should be the incrementing method called?